src/Foundation/NSDate.cs

```diff
@@ -42,10 +42,19 @@ public partial class NSDate {
 		public static explicit operator DateTime (NSDate d)
 		{
 			double secs = d.SecondsSinceReferenceDate;
-			if ((secs < -63113904000) || (secs > 252423993599))
-				throw new ArgumentOutOfRangeException ("Value is outside the range of NSDate");
 
-			return new DateTime ((long)(secs * TimeSpan.TicksPerSecond + NSDATE_TICKS), DateTimeKind.Utc);
+			if ((secs < -63114076800) || (secs > 252423993599))
+				throw new ArgumentOutOfRangeException ($"Value is outside the range of NSDate {secs}");
+
+			var calendar = new NSCalendar (NSCalendarType.Gregorian);
+			calendar.TimeZone = NSTimeZone.FromName ("UTC");
+			NSDateComponents components = calendar.Components (NSCalendarUnit.Year | NSCalendarUnit.Month | NSCalendarUnit.Day | NSCalendarUnit.Hour |
```

Comment:
Minor - `var`.

Reply:
Toss me to me. When I don't see the type on the right I don't mind it on the left. 🤷‍♂